app/services/auth/firebase_auth.py

```
@@ -0,0 +1,382 @@
+    user = session.query(User).filter(User.firebase_uid == uid).first()
+    if not user:
+        raise UserNotFound(uid)
+    # Update user details here as needed.
+    try:
+        update_data = {k: v for k, v in user_updates.model_dump(exclude={'password'}).items() if v is not None}
+        session.query(Profile).filter(Profile.user_id == user.id).update(update_data)
+        session.commit()
+    except SQLAlchemyError as e:
+        session.rollback()
+        logger.error(f"Error updating user {uid}: {e}")
+        raise InternalServerError(f"Failed to update user {uid}.") from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)